### PR TITLE
When use the custom balancer and use the defaultServiceInstace throw ClassCaseException

### PR DESCRIPTION
**Pull request by zt9788**
When use the custom balancer and use the DefaultServiceInstace throw ClassCaseException

### Describe what this PR does / why we need it
Avoiding issues caused by type casting

### Does this pull request fix one issue?
None

### Describe how you did it
None

### CODE REVIEW

**Comment by 123liuziming** [comment]
Thanks for your pr! Could you please submit an issue to record the problem in more detail and mention the issue in this pr as well?

**Comment by zt9788** [comment]
> Thanks for your pr! Could you please submit an issue to record the problem in more detail and mention the issue in this pr as well?


OK~
Related issues #3346

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/balancer/NacosBalancer.java [hunk 1]
```diff
@@ -66,30 +68,41 @@ public static ServiceInstance getHostByRandomWeight3(
...
-	private static void convertIPv4ToIPv6(NacosServiceInstance instance) {
+	private static void convertIPv4ToIPv6(ServiceInstance instance) {
+		if(!(instance instanceof NacosServiceInstance) && !(instance instanceof DefaultServiceInstance)) {
+			return;
+		}
 		if (Pattern.matches(IPV4_REGEX, instance.getHost())) {
 			String ip = instance.getMetadata().get(IPV6_KEY);
 			if (StringUtils.isNotEmpty(ip)) {
-				instance.setHost(ip);
+				if(instance instanceof NacosServiceInstance){
```
Comment (123liuziming) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/balancer/NacosBalancer.java]: The format of the code doesn't look good. Run `mvn clean install` to reformat your code.

Comment (zt9788) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/balancer/NacosBalancer.java]: OK

**Pull request closed without merging**